packages/theme/src/cli/utilities/theme-environment/proxy.ts

```diff
@@ -207,21 +217,23 @@ export function getProxyStorefrontHeaders(event: H3Event) {
   return proxyRequestHeaders
 }
 
-function proxyStorefrontRequest(event: H3Event, theme: Theme, ctx: DevServerContext) {
+function proxyStorefrontRequest(event: H3Event, ctx: DevServerContext) {
   const path = event.path.replaceAll(EXTENSION_CDN_PREFIX, '/')
   const host = event.path.startsWith(EXTENSION_CDN_PREFIX) ? 'cdn.shopify.com' : ctx.session.storeFqdn
   const url = new URL(path, `https://${host}`)
-  url.searchParams.set('preview_theme_id', String(theme.id))
-  const target = url.toString()
-
-  const proxyHeaders = getProxyStorefrontHeaders(event)
-  // Required header for CDN requests
-  proxyHeaders.referer = target
-
+  url.searchParams.set('_fd', '0')
+  url.searchParams.set('pb', '0')
```

Comment:
Turns out we had to pass `_fd` to avoid domain redirection

Reply:
Excellent finding! During my tests, in the final version of this PR, the `/localization` feature seems to work as expected with this change:

https://github.com/user-attachments/assets/d35d001d-11ba-4f50-8588-ff317bf2d321

(you might notice that query parameters are getting duplicated, but that doesn't seem related to this fix. I've raised a [separate issue](https://github.com/Shopify/develop-advanced-edits/issues/328) for that)

Reply:
That's great, I was really hoping for this to be the case but in my test shop it didn't work... probably due to some Markets configuration in the store (I'm using hydrogen-preview store, perhaps I shouldn't...).

I was originally skipping the _fd&pb from the resulting redirect but since it didn't work in my local env I removed that code. Will add it back in another PR and also do some refactoring.